### DISCUSSION TITLE
Adding conversational to SDF or SDL

### DISCUSSION THREAD

**Discussion by ronaldsholt** [sinaptik-ai/pandas-ai#494] [Q&A] [answered]
Has anyone tried to implement conversational=True to SDF or SDLs? Maybe I missed this, but is is possible to toggle this or add it as param in the config ?

**Comment by nautics889** [comment]
Hello, @ronaldsholt

Actually you can pass this flag in a config object, for example, like below:
```
sdf = SmartDataframe(df, config={"conversational": True}
```
However i couldn't found any usage of for `conversational` flag passing to the config in [current head of main](https://github.com/gventuri/pandas-ai/tree/05e7366326dc940d743ec1bd57a7aa56f53cb031). Despite there are [docsting for `consversational`](https://github.com/gventuri/pandas-ai/blob/main/pandasai/__init__.py#L123C17-L123C17) as well as [passing them to a config](https://github.com/gventuri/pandas-ai/blob/main/pandasai/__init__.py#L146) in `PandasAI` shortcut, i haven't seen any logic depending of this.
So this should do nothing yet as far as i understood. Must be some stub to later work apparently.

@gventuri correct me if i'm wrong.

**Comment by gventuri** [comment, marked as answer]
@ronaldsholt this is indeed a very good point. We are not supporting the conversational mode anymore, but a new replacement for it is coming soon. We plan to let have the `.chat` method allowing to specify whether one expect the output to be a dataframe, a chart, a number or a string (which will be what used to be the "conversational" feature). This new feature will come soon.

**Reply by rucha80** [reply]
Hi @gventuri ,

In .chat option will you be storing memory and history? So that we can connect previous question and new question?

**Reply by asharali95** [reply]
Amazing feature indeed @gventuri. However, the previous version enabled the user to interact in the form of natural language. In contrast, the newer version is missing that element. Is there alternative way to get that natural response in the current version?

**Reply by nautics889** [reply]
@rucha80 
> In .chat option will you be storing memory and history? So that we can connect previous question and new question?

Storing history sounds like a good idea for a new feature. Meanwhile pandasai has cache for optimizing purposes, but it seems different regarding to what you want to see, as i understand.
